api/socket/socket.proto

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 TII (SSRC) and the Ghaf contributors
+// SPDX-License-Identifier: Apache-2.0
+syntax = "proto3";
+option go_package = "./socketproxy";
+package socketproxy;
+
+message Empty {}
+
+message SyncData {
+    uint32 Status = 1;
+}
+
+message SocketDataStream {
```

Comment:
I guess technically this should be named `Packet` or `Frame` or similar, but up to you.